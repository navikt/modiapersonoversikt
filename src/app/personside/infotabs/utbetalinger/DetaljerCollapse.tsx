import * as React from 'react';
import { ReactNode } from 'react';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { UnmountClosed } from 'react-collapse';
import { FlexEnd } from '../../../../components/common-styled-components';
import NavFrontendChevron from 'nav-frontend-chevron';
import { Normaltekst } from 'nav-frontend-typografi';

interface Props {
    open: boolean;
    toggle: () => void;
    children: ReactNode;
}

const Wrapper = styled<{ open: boolean }, 'div'>('div')`
  transition: .5s;
  padding: 0 ${theme.margin.px20} ${theme.margin.px20};
  ${props => props.open && theme.ekspandert};
`;

const CollapseAnimasjon = styled<{ open: boolean }, 'div'>('div')`
  transition: .5s;
  padding: 0 ${theme.margin.px20};
  ${props => props.open && 'padding: 0'};
`;

const KnappWrapper = styled<{ open: boolean }, 'button'>('button')`
  border: none;
  padding: .1rem .2rem;
  margin-top: ${props => props.open && '1rem'};
  border-radius: ${theme.borderRadius.knapp};
  cursor: pointer;
  background-color: transparent;
  display: flex;
  align-items: center;
  color: ${theme.color.lenke};
  &:focus {
    ${theme.focus}
  }
  &:hover {
    opacity: 0.8;
    text-decoration: underline;
  }
 @media print {
    display: none;
 }
`;

const Luft = styled.span`
  margin-right: .5rem;
`;

interface KnappProps {
    onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
    open: boolean;
}

function DetaljerKnapp(props: KnappProps) {
    return (
        <KnappWrapper
            onClick={props.onClick}
            aria-expanded={props.open}
            open={props.open}
        >
            <Luft>
                <Normaltekst tag="span">
                    {props.open ? 'Skjul' : 'Vis'} detaljer
                </Normaltekst>
            </Luft>
            <NavFrontendChevron type={props.open ? 'opp' : 'ned'}/>
        </KnappWrapper>
    );
}

function DetaljerCollapse(props: Props) {
    return (
        <Wrapper open={props.open}>
            <CollapseAnimasjon open={props.open}>
                <UnmountClosed isOpened={props.open}>
                    {props.children}
                </UnmountClosed>
            </CollapseAnimasjon>
            <FlexEnd>
                <DetaljerKnapp
                    onClick={props.toggle}
                    open={props.open}
                />
            </FlexEnd>
        </Wrapper>
    );
}

export default DetaljerCollapse;
