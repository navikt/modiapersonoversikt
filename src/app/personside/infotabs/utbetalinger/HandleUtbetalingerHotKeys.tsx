import * as React from 'react';
import { Utbetaling, Ytelse } from '../../../../models/utbetalinger';
import { flatMapYtelser } from './utils/utbetalingerUtils';
import { AnyAction, Dispatch } from 'redux';
import {
    setEkspanderYtelse,
    setNyYtelseIFokus
} from '../../../../redux/utbetalinger/utbetalingerStateReducer';
import { AppState } from '../../../../redux/reducers';
import { connect } from 'react-redux';
import styled from 'styled-components';

interface OwnProps {
    children: React.ReactNode;
    utbetalinger: Utbetaling[];
}

interface DispatchProps {
    settYtelseIFokus: (ytelse: Ytelse | null) => void;
    ekspanderYtelse: (ytelse: Ytelse, ekspander: boolean) => void;
}

interface StateProps {
    ytelseIFokus: Ytelse | null;
    ekspanderteYtelser: Ytelse[];
}

type Props = DispatchProps & OwnProps & StateProps;

function eventTargetIsButton(event: React.KeyboardEvent) {
    return event.target instanceof HTMLButtonElement;
}

const Wrapper = styled.div`
  &:focus {
    outline: none;
  }
`;

class HandleUtbetalingerHotKeys extends React.Component<Props> {

    constructor(props: Props) {
        super(props);
        this.handleKeyDown = this.handleKeyDown.bind(this);
    }

    handleKeyDown(event: React.KeyboardEvent) {
        switch (event.which) {
            case 13:
                this.handleEnter(event);
                break;
            case 32:
                this.handleSpace(event);
                break;
            case 38:
                this.settForrigeYtelseIFokus();
                break;
            case 40:
                this.settNesteYtelseIFokus();
                break;
            default:
        }
    }

    settNesteYtelseIFokus() {
        const ytelser: Ytelse[] = flatMapYtelser(this.props.utbetalinger);
        const currentIndex = this.props.ytelseIFokus ? ytelser.indexOf(this.props.ytelseIFokus) : -1;
        const nesteYtelse = ytelser[currentIndex + 1] || ytelser[0];
        this.props.settYtelseIFokus(nesteYtelse);
    }

    settForrigeYtelseIFokus() {
        const ytelser: Ytelse[] = flatMapYtelser(this.props.utbetalinger);
        const currentIndex = this.props.ytelseIFokus ? ytelser.indexOf(this.props.ytelseIFokus) : ytelser.length;
        const forrigeYtelse = ytelser[currentIndex - 1] || ytelser[ytelser.length - 1];
        this.props.settYtelseIFokus(forrigeYtelse);
    }

    handleEnter(event: React.KeyboardEvent) {
        if (event.repeat || eventTargetIsButton(event)) {
            return;
        }
        this.toggleEkspanderYtelseIFokus();
    }

    toggleEkspanderYtelseIFokus() {
        const ytelseIFokus = this.props.ytelseIFokus;
        if (!ytelseIFokus) {
            return;
        }
        const erEkspandert = this.props.ekspanderteYtelser.includes(ytelseIFokus);
        this.props.ekspanderYtelse(ytelseIFokus, !erEkspandert);
    }

    handleSpace(event: React.KeyboardEvent) {
        if (eventTargetIsButton(event)) {
            return;
        }
        event.preventDefault();
        this.toggleEkspanderYtelseIFokus();
    }

    render() {
        return (
            <Wrapper onKeyDown={this.handleKeyDown} tabIndex={-1}>
                {this.props.children}
            </Wrapper>
        );
    }
}

function mapDispatchToProps(dispatch: Dispatch<AnyAction>): DispatchProps {
    return {
        settYtelseIFokus: ytelse => dispatch(setNyYtelseIFokus(ytelse)),
        ekspanderYtelse: (ytelse: Ytelse, ekspander: boolean) => dispatch(setEkspanderYtelse(ytelse, ekspander))
    };
}

function mapStateToProps(state: AppState): StateProps {
    return {
        ytelseIFokus: state.utbetalinger.ytelseIFokus,
        ekspanderteYtelser: state.utbetalinger.ekspanderteYtelser
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(HandleUtbetalingerHotKeys);
