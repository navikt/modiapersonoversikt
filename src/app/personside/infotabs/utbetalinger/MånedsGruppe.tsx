import * as React from 'react';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { ArrayGroup } from '../../../../utils/groupArray';
import { Normaltekst } from 'nav-frontend-typografi';
import { Bold, Uppercase } from '../../../../components/common-styled-components';
import { getGjeldendeDatoForUtbetaling } from './utils/utbetalingerUtils';
import UtbetalingsKomponent from './utbetaling/Utbetaling';
import { Utbetaling } from '../../../../models/utbetalinger';

interface Props {
    gruppe: ArrayGroup<Utbetaling>;
}

const Wrapper = styled.li`
  > *:first-child {
    background-color: ${theme.color.kategori};
    padding: .2rem ${theme.margin.px20};
  }
  ol {
    padding: 0;
    margin: 0;
  }
  ol > *:not(:first-child) {
    border-top: ${theme.border.skille};
  }
`;

function Månedsgruppe({gruppe}: Props) {
    const utbetalingsKomponenter = gruppe.array.map(utbetaling => (
        <UtbetalingsKomponent
            key={getGjeldendeDatoForUtbetaling(utbetaling) + utbetaling.nettobeløp}
            utbetaling={utbetaling}
        />
    ));
    return (
        <Wrapper>
            <Normaltekst tag={'h3'}><Bold><Uppercase>{gruppe.category}</Uppercase></Bold></Normaltekst>
            <ol>
                {utbetalingsKomponenter}
            </ol>
        </Wrapper>
    );
}

export default Månedsgruppe;
