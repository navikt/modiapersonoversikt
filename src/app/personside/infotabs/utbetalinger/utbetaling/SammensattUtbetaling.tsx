import * as React from 'react';
import { formaterNOK, getGjeldendeDatoForUtbetaling, getNettoSumYtelser } from '../utils/utbetalingerUtils';
import styled from 'styled-components';
import { Bold, FlexEnd, SpaceBetween } from '../../../../../components/common-styled-components';
import PrintKnapp from '../../../../../components/PrintKnapp';
import { Normaltekst } from 'nav-frontend-typografi';
import { Utbetaling, Ytelse } from '../../../../../models/utbetalinger';
import theme from '../../../../../styles/personOversiktTheme';
import DelUtbetaling from './DelUtbetaling';
import Printer from '../../../../../utils/Printer';
import { loggEvent } from '../../../../../utils/frontendLogger';
import { connect } from 'react-redux';
import { UtbetalingTabellStyling } from '../utils/CommonStyling';
import { AnyAction, Dispatch } from 'redux';
import { setEkspanderYtelse } from '../../../../../redux/utbetalinger/actions';
import { datoVerbose } from '../../../../../utils/dateUtils';

interface OwnProps {
    utbetaling: Utbetaling;
}

interface DispatchProps {
    ekspanderYtelse: (ytelse: Ytelse) => void;
}

type Props = DispatchProps & OwnProps;

const SammensattUtbetalingStyle = styled.li`
    padding: ${theme.margin.px20};
    > *:nth-child(3) {
        margin-bottom: 0.8rem;
    }
    @media print {
        list-style-type: none;
        li {
            page-break-inside: avoid;
        }
    }
`;

const YtelsesListe = styled.ul`
    margin-top: 2rem;
    padding: 0;
    list-style: none;
    border-radius: ${theme.borderRadius.layout};
    border: ${theme.border.skilleSvak};
    @media print {
        border: none;
    }
    > li:not(:first-child) {
        border-top: ${theme.border.skilleSvak};
    }
`;

class SammensattUtbetaling extends React.PureComponent<Props> {
    private print?: () => void;

    constructor(props: Props) {
        super(props);
        this.visDetaljerAndPrint = this.visDetaljerAndPrint.bind(this);
    }

    visDetaljerAndPrint() {
        const ytelser = this.props.utbetaling.ytelser;

        if (!ytelser) {
            return;
        }
        ytelser.forEach(ytelse => this.props.ekspanderYtelse(ytelse));
        this.print && this.print();
        loggEvent('SammensattUtbetaling', 'Printer');
    }

    render() {
        const utbetaling = this.props.utbetaling;

        if (!utbetaling.ytelser) {
            return <>'Manglende data i utbetaling.'</>;
        }

        const dato = datoVerbose(getGjeldendeDatoForUtbetaling(utbetaling)).sammensatt;
        const sum = formaterNOK(getNettoSumYtelser(utbetaling.ytelser));
        const forfallsInfo = utbetaling.forfallsdato && !utbetaling.utbetalingsdato ? `Forfallsdato: ${dato}` : '';
        const ytelsesListe = utbetaling.ytelser.map((ytelse, index) => (
            <DelUtbetaling ytelse={ytelse} konto={utbetaling.konto} melding={utbetaling.melding} key={index} />
        ));

        return (
            <Printer getPrintTrigger={(trigger: () => void) => (this.print = trigger)}>
                <SammensattUtbetalingStyle>
                    <UtbetalingTabellStyling>
                        <Normaltekst>
                            {dato} / <Bold>{utbetaling.status}</Bold>
                        </Normaltekst>
                        <SpaceBetween>
                            <Normaltekst tag={'h4'}>
                                <Bold>Diverse ytelser</Bold>
                            </Normaltekst>
                            <Normaltekst>
                                <Bold>{sum}</Bold>
                            </Normaltekst>
                        </SpaceBetween>
                        <FlexEnd>
                            <Normaltekst>{forfallsInfo}</Normaltekst>
                        </FlexEnd>
                        <SpaceBetween>
                            <Normaltekst>Utbetaling til: {utbetaling.utbetaltTil}</Normaltekst>
                            <PrintKnapp onClick={this.visDetaljerAndPrint} />
                        </SpaceBetween>
                        <YtelsesListe aria-label={`Diverse ytelser`}>{ytelsesListe}</YtelsesListe>
                    </UtbetalingTabellStyling>
                </SammensattUtbetalingStyle>
            </Printer>
        );
    }
}

export default connect(
    null,
    (dispatch: Dispatch<AnyAction>): DispatchProps => {
        return {
            ekspanderYtelse: ytelse => dispatch(setEkspanderYtelse(ytelse, true))
        };
    }
)(SammensattUtbetaling);
