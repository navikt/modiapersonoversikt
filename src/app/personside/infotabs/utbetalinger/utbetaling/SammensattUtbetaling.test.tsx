import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SammensattUtbetaling from './SammensattUtbetaling';
import { Utbetaling } from '../../../../../models/utbetalinger';
import { statiskMockUtbetaling, statiskMockYtelse } from '../../../../../mock/statiskMockUtbetaling';
import TestProvider from '../../../../../test/Testprovider';

// Mock react collapse sin UnmountClosed
jest.mock('react-collapse', () => {
    return {
        // @ts-ignore
        UnmountClosed: props => props.children
    };
});

const sammensattUtbetaling: Utbetaling = {
    ...statiskMockUtbetaling,
    ytelser: [statiskMockYtelse, statiskMockYtelse]
};

test('Viser utbetalingsdetaljer riktig med liste med ytelser og trekk', () => {
    const visittkortheader = renderer.create(
        <TestProvider>
            <SammensattUtbetaling utbetaling={sammensattUtbetaling} />
        </TestProvider>
    );

    const json = visittkortheader.toJSON();
    expect(json).toMatchSnapshot();
});
