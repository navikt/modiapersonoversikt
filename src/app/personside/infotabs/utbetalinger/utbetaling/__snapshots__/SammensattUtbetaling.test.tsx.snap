// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser utbetalingsdetaljer riktig med liste med ytelser og trekk 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  font-weight: bold;
}

.c9 {
  position: relative;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  padding-left: 1.5rem;
}

.c9::before {
  position: absolute;
  left: -0.5rem;
  content: 'â€¢';
  font-size: 4rem;
  line-height: 1.8rem;
  color: #cce1f3;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c12 {
  position: relative;
  border: none;
  padding: 0;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0067c5;
}

.c12:after {
  border-bottom: 1px #b7b1a9 solid;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
}

.c12:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c12:hover {
  opacity: 0.8;
}

.c12:hover:after {
  content: '';
  border-color: #0067c5;
}

.c5 {
  border: none;
  background-color: transparent;
  padding: 0.1rem 0.2rem;
  border-radius: .35rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #0067c5;
}

.c5 svg {
  margin-left: 0.5rem;
  height: 1.2rem;
  width: 1.2rem;
}

.c5:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c5:hover {
  opacity: 0.8;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c15 th:not(:first-child) {
  font-weight: normal;
  font-style: italic;
}

.c16 td:not(:first-child) {
  font-weight: bold;
}

.c17 {
  border-top: solid 0.0625rem #c6c2bf;
  margin: 0.5rem 0;
}

.c13 {
  margin-right: 0.5rem;
}

.c8 {
  -webkit-transition: 0.5s;
  transition: 0.5s;
  padding: 1.25rem;
}

.c14 {
  -webkit-transition: 0.5s;
  transition: 0.5s;
  padding: 0 1.25rem;
}

.c10 {
  margin-bottom: 1.25rem;
}

.c7 {
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c1 table {
  width: 100%;
  text-align: right;
  border-spacing: 0;
}

.c1 table * {
  padding: 0;
  margin: 0;
}

.c1 table tr > * {
  padding: 0.1rem;
  text-align: right;
}

.c1 table tr > *:first-child {
  text-align: left;
}

.c1 table tr > *:not(:first-child) {
  width: 6rem;
}

.c0 {
  padding: 1.25rem;
}

.c0 > *:nth-child(3) {
  margin-bottom: 0.8rem;
}

.c6 {
  margin-top: 2rem;
  padding: 0;
  list-style: none;
  border-radius: .25rem;
  border: solid 0.0625rem #c6c2bf;
}

.c6 > li:not(:first-child) {
  border-top: solid 0.0625rem #c6c2bf;
}

@media print {
  .c5 {
    display: none;
  }
}

@media print {
  .c11 {
    display: none;
  }
}

@media print {
  .c0 {
    list-style-type: none;
  }

  .c0 li {
    page-break-inside: avoid;
  }
}

@media print {
  .c6 {
    border: none;
  }
}

<li
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="typo-normal"
    >
      29. Mars 2017
       / 
      <span
        className="c2"
      >
        Utbetalt
      </span>
    </p>
    <div
      className="c3"
    >
      <h4
        className="typo-normal"
      >
        <span
          className="c2"
        >
          Diverse ytelser
        </span>
      </h4>
      <p
        className="typo-normal"
      >
        <span
          className="c2"
        >
          22,772.00
        </span>
      </p>
    </div>
    <div
      className="c4"
    >
      <p
        className="typo-normal"
      >
        
      </p>
    </div>
    <div
      className="c3"
    >
      <p
        className="typo-normal"
      >
        Utbetaling til: 
        AREMARK TESTFAMILIEN
      </p>
      <button
        aria-label="Skriv ut"
        className="c5"
        onClick={[Function]}
      >
        <span
          className="typo-normal"
        >
          Skriv ut
        </span>
         
        <svg
          viewBox="0 0 17 16"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M14.8,4.9H2.2c-0.9,0-1.7,0.7-1.7,1.7v4c0,0.9,0.7,1.7,1.7,1.7h1.7v3c0,0.2,0.1,0.3,0.3,0.3h8.7	c0.2,0,0.3-0.1,0.3-0.3v-3h1.7c0.9,0,1.7-0.7,1.7-1.7v-4C16.5,5.6,15.8,4.9,14.8,4.9z M4.5,14.9h8v-4.7h-8V14.9z"
            fill="#0067C5"
            fillRule="evenodd"
          />
          <path
            d="M4.9,0.9h7.1c0.6,0,1,0.4,1,1v1.4c0,0.6-0.4,1-1,1H4.9c-0.6,0-1-0.4-1-1V1.9C3.9,1.3,4.4,0.9,4.9,0.9z"
            fill="#0067C5"
          />
        </svg>
      </button>
    </div>
    <ul
      aria-label="Diverse ytelser"
      className="c6"
    >
      <li
        className="c7"
        onClick={[Function]}
        onFocus={[Function]}
        tabIndex={0}
      >
        <article
          aria-expanded={false}
          aria-label="Delutbetaling Sykepenger"
        >
          <div
            className="c8"
            open={false}
          >
            <div
              className="c9"
              color="#cce1f3"
            >
              <div
                className="c3"
              >
                <h5
                  className="typo-normal"
                >
                  <span
                    className="c2"
                  >
                    Sykepenger
                  </span>
                </h5>
                <p
                  className="typo-normal"
                >
                  <span
                    className="c2"
                  >
                    11,386.00
                  </span>
                </p>
              </div>
              <p
                className="typo-normal"
              >
                09.11.2016 - 31.10.2017
              </p>
            </div>
            <div
              className="c10"
            >
              <div
                className="c11"
              >
                <div
                  className="c4"
                >
                  <button
                    aria-expanded={false}
                    className="c12"
                    onClick={[Function]}
                  >
                    <span
                      className="c13"
                    >
                      <span
                        className="typo-normal"
                      >
                        Vis
                         
                        detaljer
                      </span>
                    </span>
                    <i
                      className="nav-frontend-chevron chevronboks chevron--ned"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              className="c14"
              open={false}
            >
              <aside
                aria-label="Utbetalingsdetaljer"
                className="c15"
              >
                <span
                  className="typo-normal"
                >
                  <div
                    className="c16"
                  >
                    <table
                      role="table"
                    >
                      <thead
                        role="rowgroup"
                      >
                        <tr
                          role="row"
                        >
                          <th
                            role="columnheader"
                          >
                            Konto
                          </th>
                          <th
                            role="columnheader"
                          >
                            Brutto
                          </th>
                          <th
                            role="columnheader"
                          >
                            Trekk
                          </th>
                          <th
                            role="columnheader"
                          >
                            Utbetalt
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        role="rowgroup"
                      >
                        <tr
                          role="row"
                        >
                          <td
                            role="cell"
                          >
                            83980854563
                          </td>
                          <td
                            role="cell"
                          >
                            13,370.00
                          </td>
                          <td
                            role="cell"
                          >
                            -1,984.00
                          </td>
                          <td
                            role="cell"
                          >
                            11,386.00
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </span>
                <div
                  className="c17"
                />
                <span
                  className="typo-normal"
                >
                  <table
                    role="table"
                  >
                    <thead
                      role="rowgroup"
                    >
                      <tr
                        role="row"
                      >
                        <th
                          role="columnheader"
                        >
                          Detaljer
                        </th>
                        <th
                          role="columnheader"
                        >
                          Sats
                        </th>
                        <th
                          role="columnheader"
                        >
                          Antall
                        </th>
                        <th
                          role="columnheader"
                        >
                          BelÃ¸p
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      role="rowgroup"
                    >
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Bacon
                        </td>
                        <td
                          role="cell"
                        >
                          558.00
                        </td>
                        <td
                          role="cell"
                        >
                          10
                        </td>
                        <td
                          role="cell"
                        >
                          5,580.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Computer
                        </td>
                        <td
                          role="cell"
                        >
                          1,310.00
                        </td>
                        <td
                          role="cell"
                        >
                          3
                        </td>
                        <td
                          role="cell"
                        >
                          3,930.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Mouse
                        </td>
                        <td
                          role="cell"
                        >
                          1,930.00
                        </td>
                        <td
                          role="cell"
                        >
                          2
                        </td>
                        <td
                          role="cell"
                        >
                          3,860.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Skattetrekk
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          -730.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Prosenttrekk
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          -1,042.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Prosenttrekk
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          -212.00
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </span>
                <div
                  className="c17"
                />
                <h4
                  className="typo-normal"
                >
                  <span
                    className="c2"
                  >
                    Melding
                  </span>
                </h4>
                <p
                  className="typo-normal"
                >
                  Utbetalingsmelding
                </p>
              </aside>
            </div>
          </div>
        </article>
      </li>
      <li
        className="c7"
        onClick={[Function]}
        onFocus={[Function]}
        tabIndex={0}
      >
        <article
          aria-expanded={false}
          aria-label="Delutbetaling Sykepenger"
        >
          <div
            className="c8"
            open={false}
          >
            <div
              className="c9"
              color="#cce1f3"
            >
              <div
                className="c3"
              >
                <h5
                  className="typo-normal"
                >
                  <span
                    className="c2"
                  >
                    Sykepenger
                  </span>
                </h5>
                <p
                  className="typo-normal"
                >
                  <span
                    className="c2"
                  >
                    11,386.00
                  </span>
                </p>
              </div>
              <p
                className="typo-normal"
              >
                09.11.2016 - 31.10.2017
              </p>
            </div>
            <div
              className="c10"
            >
              <div
                className="c11"
              >
                <div
                  className="c4"
                >
                  <button
                    aria-expanded={false}
                    className="c12"
                    onClick={[Function]}
                  >
                    <span
                      className="c13"
                    >
                      <span
                        className="typo-normal"
                      >
                        Vis
                         
                        detaljer
                      </span>
                    </span>
                    <i
                      className="nav-frontend-chevron chevronboks chevron--ned"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              className="c14"
              open={false}
            >
              <aside
                aria-label="Utbetalingsdetaljer"
                className="c15"
              >
                <span
                  className="typo-normal"
                >
                  <div
                    className="c16"
                  >
                    <table
                      role="table"
                    >
                      <thead
                        role="rowgroup"
                      >
                        <tr
                          role="row"
                        >
                          <th
                            role="columnheader"
                          >
                            Konto
                          </th>
                          <th
                            role="columnheader"
                          >
                            Brutto
                          </th>
                          <th
                            role="columnheader"
                          >
                            Trekk
                          </th>
                          <th
                            role="columnheader"
                          >
                            Utbetalt
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        role="rowgroup"
                      >
                        <tr
                          role="row"
                        >
                          <td
                            role="cell"
                          >
                            83980854563
                          </td>
                          <td
                            role="cell"
                          >
                            13,370.00
                          </td>
                          <td
                            role="cell"
                          >
                            -1,984.00
                          </td>
                          <td
                            role="cell"
                          >
                            11,386.00
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </span>
                <div
                  className="c17"
                />
                <span
                  className="typo-normal"
                >
                  <table
                    role="table"
                  >
                    <thead
                      role="rowgroup"
                    >
                      <tr
                        role="row"
                      >
                        <th
                          role="columnheader"
                        >
                          Detaljer
                        </th>
                        <th
                          role="columnheader"
                        >
                          Sats
                        </th>
                        <th
                          role="columnheader"
                        >
                          Antall
                        </th>
                        <th
                          role="columnheader"
                        >
                          BelÃ¸p
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      role="rowgroup"
                    >
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Bacon
                        </td>
                        <td
                          role="cell"
                        >
                          558.00
                        </td>
                        <td
                          role="cell"
                        >
                          10
                        </td>
                        <td
                          role="cell"
                        >
                          5,580.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Computer
                        </td>
                        <td
                          role="cell"
                        >
                          1,310.00
                        </td>
                        <td
                          role="cell"
                        >
                          3
                        </td>
                        <td
                          role="cell"
                        >
                          3,930.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Mouse
                        </td>
                        <td
                          role="cell"
                        >
                          1,930.00
                        </td>
                        <td
                          role="cell"
                        >
                          2
                        </td>
                        <td
                          role="cell"
                        >
                          3,860.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Skattetrekk
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          -730.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Prosenttrekk
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          -1,042.00
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          role="cell"
                        >
                          Prosenttrekk
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          
                        </td>
                        <td
                          role="cell"
                        >
                          -212.00
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </span>
                <div
                  className="c17"
                />
                <h4
                  className="typo-normal"
                >
                  <span
                    className="c2"
                  >
                    Melding
                  </span>
                </h4>
                <p
                  className="typo-normal"
                >
                  Utbetalingsmelding
                </p>
              </aside>
            </div>
          </div>
        </article>
      </li>
    </ul>
  </div>
</li>
`;
